Allow async functions in the Vercel edge middleware integration (#9334)
withastro/astro: .changeset/young-spoons-rescue.md

---

> '@astrojs/vercel': minor

Reviewer: ```suggestion
'@astrojs/vercel': patch
```
A minor release usually means there is something new for users to read up on.

Author: Well this does allow for you to return a Promise from the middleware and you previously couldn't do that. So, as long the version increases somewhere, I think it's fine.

Reviewer 2: What Arsh means is that the version will be increased automatically by a bot in an future PR, so you only have to specify the change type now

Author: I'm just not sure what or if the version will change when the changeset is set to a patch. Will it theoretically go from 6.0.0 => 6.0.1?

Reviewer 2: I think another patch has been released so it's already 6.0.1 but yeah, in a release PR, it will be bumped to 6.0.2 as well as other packages depending on it

---

> Allow async functions in the Vercel edge middleware integration.

Reviewer: ```suggestion
Allows the edge middleware to be an async function.
```

> Allow async functions in the Vercel edge middleware integration.

Reviewer: I did't see this but would you believe I made the exact same suggestion!

> Allow async functions in the Vercel edge middleware integration.

Reviewer: Just had an extra "the".
